summary_covidseq.R

```diff
@@ -0,0 +1,133 @@
+ReadType<-xmlToDataFrame(nodes=getNodeSet(xmlParse(read_xml(fileName)),"//ReadType"));ReadType
+ReadLength<-xmlToDataFrame(nodes=getNodeSet(xmlParse(read_xml(fileName)),"//Read1NumberOfCycles"));ReadLength 
+
+#=============================================
+# Number of PatientSample (n); 
+df1<-tail(df1,-13);
+df1<-df1[,c(1,3)]; 
+colnames(df1) <- c("Sample_Accession","Sample_Type");
+df1<-filter(df1, Sample_Type == "PatientSample"); 
+n<-length(df1$Sample_Type); cat("Number of PatientSamples",n,"\n");Sys.sleep(t)
+
+#=============================================
+# Random Number Generator for Sample_ID
+RNG<-paste(floor(runif(n, min=99, max=1000)),floor(runif(n, min=99, max=1000)),sep =""); 
+
+#============================================
+# Checking for duplicates in RNG
+while(length(RNG)!= length(unique(RNG)))
+     { RNG<-paste(floor(runif(n, min=99, max=1000)),floor(runif(n, min=99, max=1000)),sep ="");}
+IDD<-paste("UT-UPHL",substr(args[1], 11, 16),sep="-");
+Sample_ID<-paste(IDD,RNG,sep=""); 
+summary1<-cbind(df1,Sample_ID);
+head(summary1)
+
+#============================================
+#left join: SampleSheet & Combined Lineage Report files
+var <-paste('',args[1],sep='-'); head(var)
+df2 <-df2 %>% mutate_at("taxon", str_replace, var, "")
+df2<-df2[,c("taxon","lineage","scorpio_call")]
+head(df2); #Taxon, lineage, scorpio_call
+df3 = read.table(file = ss3, sep = '\t', header = TRUE); # 
+df3<-df3 %>% mutate_at("Sample", str_replace, var, ""); head(df3)
+df3<-df3[,c("Sample","nTargetsDetected.SARS.CoV2")]; 
+colnames(df3)[2] <- 'sc2_amplicons';#
+summary1<-merge(summary1,df2, by.x="Sample_Accession", by.y= "taxon", all.x=TRUE); 
+head(summary1)
+summary1<-merge(summary1,df3, by.x="Sample_Accession", by.y= "Sample", all.x=TRUE);
+
+#==========================================
+# adding generalc information: 
+summary1<-cbind(summary1,'wetlab_assay'= wetlab_assay,"instrument_type"=instrument_type,"software"=software,"lineage_tools"=lineage_tools,
+                "pangolin_version"=pangolin_version,'read_Type' = ReadType,"readLength" = ReadLength)
+                colnames(summary1)[12]<-("Read_Type");
+                colnames(summary1)[13]<-("Read_Length");
+                head(summary1)
+
+#==========================================
+# Nucleotides count: (num_actg & num_n
+num_actgx<-vector();num_nx<-vector(); pass_fail<-vector()
+runPath1 = paste(runPath,"covidseq_output/Sample_Analysis/",sep ="/"); 
+num_actgx = 0; num_nx = 0;
+for (i in 1:n){
+     sample=summary1$Sample_Accession[i]; 
+     amplicons=summary1$sc2_amplicons[i];
+      if(amplicons >=50)
+      { 
+        Lyapunovv<- Sys.glob(paste(runPath1,sample,"*/*.fasta",sep = ""),dirmark = TRUE); 
+        a =  str_count(read.fasta(Lyapunovv, as.string = T, forceDNAtolower = T), pattern = "a|c|t|g");num_actgx[i] = a;
+        b =  str_count(read.fasta(Lyapunovv, as.string = T, forceDNAtolower = T))-a; num_nx[i] = b; pass_fail[i] ="pass"; c = "pass"
```

Comment:
Just to give myself a bit of sanity, I ran this through `/Volumes/NGS/Analysis/covidseq/UT-A01290-220606/covidseq_output/Sample_Analysis/2864876-UT-A01290-220606/2864876-UT-A01290-220606.fasta`.

This file has 8593 a's, 5238 c's, 9263 t's, and 5623 g's outside of the header.

Using 
```
print(str_count(read.fasta(fasta, as.string = T, forceDNAtolower = T), pattern = "a"))
```
I observe 8593 a's, 5238 c's, 9263 t's, and 5623 g's. 

Which is the same as counting the characters. In summary, read.fasta doesn't include the header. (All samples have a T in their name `>2864876-UT-A01290-220606`, so I was worried.)